Author: Fabrice Fontaine <https://github.com/ffontaine>, Andrew Harle <46223597+andrewharle@users.noreply.github.com>
Description: Cherry pick patch from https://github.com/mongodb/mongo/pull/1296/commits/362be06fc16a5ad0f9e9aa90cc763c5242e8e35c to allow build with 32-bit SSL on armhf.

Last-Update: 2019-08-18

--- mongodb-4.0.12.orig/src/mongo/util/net/ssl_manager.cpp
+++ mongodb-4.0.12/src/mongo/util/net/ssl_manager.cpp
@@ -410,7 +410,7 @@ public:
      *
      * Returns a DERToken which consists of the (tag, length, value) tuple.
      */
-    static StatusWith<DERToken> parse(ConstDataRange cdr, size_t* outLength);
+    static StatusWith<DERToken> parse(ConstDataRange cdr, uint64_t* outLength);
 
 private:
     DERType _type{DERType::EndOfContent};
@@ -427,7 +427,7 @@ struct DataType::Handler<DERToken> {
                        size_t length,
                        size_t* advanced,
                        std::ptrdiff_t debug_offset) {
-        size_t outLength;
+        uint64_t outLength;
 
         auto swPair = DERToken::parse(ConstDataRange(ptr, length), &outLength);
 
@@ -472,7 +472,7 @@ StatusWith<std::string> readDERString(Co
 }
 
 
-StatusWith<DERToken> DERToken::parse(ConstDataRange cdr, size_t* outLength) {
+StatusWith<DERToken> DERToken::parse(ConstDataRange cdr, uint64_t* outLength) {
     const size_t kTagLength = 1;
     const size_t kTagLengthAndInitialLengthByteLength = kTagLength + 1;
 
